code/src/layouts/header/components/header-toolbar/HeaderToolbar.tsx

```diff
@@ -56,19 +64,50 @@ const HeaderToolbar = () => {
     <AppTypography variant="caption-small">{itemsInCartCount}</AppTypography>
   );
 
-  const AuthButton = isAuthenticated ? (
-    <AppButton onClick={handleLogout} variant="danger" size="small">
+  const loadingButton = isLoadingAuth ? <AppLoader /> : null;
+
+  const logoutButton = isAuthenticated ? (
+    <AppButton
+      onClick={handleLogout}
+      variant="danger"
+      size="small"
+      data-testid="LogoutButton"
+    >
       <LogoutIcon />
     </AppButton>
-  ) : (
-    <AppButton onClick={handleOpenAuthModal}>
-      <AppTypography translationKey="signIn.label" />
-    </AppButton>
-  );
+  ) : null;
 
-  const OrdersButton = isAuthenticated && (
-    <AppIconButton to={routes.orders.path} component={AppLink}>
-      <ListAltIcon className="header__toolbar-icon" fontSize="large" />
+  const signInButton =
+    !isLoadingAuth && !isAuthenticated ? (
+      <AppButton onClick={handleOpenAuthModal}>
+        <AppTypography translationKey="signIn.label" />
+      </AppButton>
+    ) : null;
+
+  const authButton = loadingButton || logoutButton || signInButton;
+
+  const loadingOrdersButton = isLoadingAuth ? <AppLoader /> : null;
+
+  const authenticatedOrdersButton =
+    isAuthenticated && !isLoadingAuth ? (
+      <AppIconButton to={routes.orders.path} component={AppLink}>
+        <ListAltIcon className="header__toolbar-icon" fontSize="large" />
+      </AppIconButton>
+    ) : null;
```

Comment:
Same here. Maybe by using `&&` we will make code cleaner

Reply:
> Same here. Maybe by using `&&` we will make code cleaner

Sure. It's look better. Fixed